spec/cucumber/core/report/summary_spec.rb

```diff
@@ -65,6 +65,18 @@ module Cucumber::Core::Report
         expect( @summary.test_cases.total(:flaky) ).to eq(1)
         expect( @summary.test_cases.total ).to eq(1)
       end
+
+      it "handless flaky with following skip test cases" do
+        allow(test_case).to receive(:==).and_return(false, false, true)
```

Comment:
I think this is telling the summary report that there are two test cases: because only the third event returns true here, so only one the last event reflects a retry.